cpp/oneapi/dal/algo/knn/backend/gpu/infer_kernel_impl_dpc_distr.hpp

```diff
@@ -699,26 +647,28 @@ sycl::event bf_kernel_distr(sycl::queue& queue,
             next_event = search(query, callback, qbcount, curr_k, { next_event });
         }
 
-        auto send_count = current_block.get_count();
-        ONEDAL_ASSERT(send_count >= 0);
-        ONEDAL_ASSERT(send_count <= de::limits<int>::max());
-        // send recv replace
-        comm.sendrecv_replace(array<Float>::wrap(queue,
-                                                 current_block.get_mutable_data(),
-                                                 send_count,
-                                                 { next_event }),
-                              prev_node,
-                              next_node)
-            .wait();
-        train_block_queue.emplace_back(current_block);
-        comm.sendrecv_replace(array<res_t>::wrap(queue,
-                                                 current_tresps.get_mutable_data(),
-                                                 current_tresps.get_count(),
-                                                 { next_event }),
-                              prev_node,
-                              next_node)
-            .wait();
-        tresps_queue.emplace_back(current_tresps);
+        if (relative_block_idx < block_count - 1) {
+            ONEDAL_PROFILER_TASK(distributed_loop.sendrecv_replace, queue);
+            auto send_count = current_block.get_count();
+            ONEDAL_ASSERT(send_count >= 0);
+            ONEDAL_ASSERT(send_count <= de::limits<int>::max());
```

Comment:
```suggestion
            ONEDAL_ASSERT(0 <= send_count);
            ONEDAL_ASSERT(send_count <= de::limits<int>::max());
```

Reply:
Are you sure that it is exactly `int`?

Reply:
I believe during testing with initial implementation there were problems if transfer size was greater than max(int32). However I haven't encountered issues lately, and large benchmarks run with high accuracy. I can check if it can be removed.

Reply:
Also I think having send_count on the left in both lines makes it more readable, so will probably leave as is unless its for uniformity - but I see other uses of >= in repo